hook.sh

```diff
@@ -46,17 +46,40 @@ echo -e "\e[32mSetting challenge to ${CERTBOT_VALIDATION} ...\e[0m"
 domain=.$CERTBOT_DOMAIN
 infix=${domain%.$DEDYN_NAME}
 
+# Remove leading wildcard from infix, if present
+# *.foobar.dedyn.io gives "" while *.a.foobar.dedyn.io gives ".a"
+infix=${infix#.\*}
+
 args=( \
-    '-Ss' \
+    '-sSLf' \
     '-H' "Authorization: Token $DEDYN_TOKEN" \
     '-H' 'Accept: application/json' \
     '-H' 'Content-Type: application/json' \
-    '-d' '[{"subname":"_acme-challenge'"$infix"'", "type":"TXT", "records":["\"'"$CERTBOT_VALIDATION"'\""], "ttl":60}]' \
-    '-o' '/dev/null' \
 )
 
+# Letsencrypt uses a TTL of 60 seconds on their resolvers. If we're getting
+# wildcard certificates (e.g. foobar.dedyn.io and *.foobar.dedyn.io), multiple
+# certificates for the same domain (e.g. RSA and ECDSA), or both, we'll need to
+# wait 60 seconds for the records to expire before verification begins. We'll
+# store any existing challenges and publish them along with the current one.
+# Set timeout_ttl=yes in ./dedynauth if you want to enable this.
```

Comment:
The last word (`this`) is a bit unclear. I know it refers to the 60 second wait, but a reader may think it is about wildcards or multi-algorithm use.

Reply:
As above.